x-pack/plugins/security_solution/server/endpoint/lib/artifacts/task.ts

```diff
@@ -158,6 +158,7 @@ export class ManifestTask {
       if (deleteErrors.length) {
         reportErrors(this.logger, deleteErrors);
       }
+      await manifestManager.cleanup(newManifest);
```

Comment:
so, does this only perform cleanup if everything above was successful (i.e., didn't throw)?

Do we want that to be the case, or is cleanup after a failure okay too?

It seems like `cleanup()` itself is a giant `try-catch` body, that perhaps swallows the error unconditionally. So does this line need to be inside this `try`? Can we move it out if it swallows its own errors? Maybe.. _gasp_ is this a time to actually use a `finally` clause?

Reply:
Yeah. My idea was call this cleanup only if the new manifest has been created correctly, so then we can clean the `.fleet-artifacts` index taking the artifacts of the new manifest as reference.
About the try/catch and the `finally` clause, for me, at least, it doesn't make sense to have a `finally` clause as this is something should be done at the end when all the other actions have finished successfully, and not when there is an error, for example, creating the manifest.